packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -713,3 +713,18 @@ class UiKitViewController {
     await SystemChannels.platform_views.invokeMethod<void>('dispose', id);
   }
 }
+
+/// A helper class to help implementing a new platform view.
+abstract class PlatformViewController {
+
+  /// This is invoked when the Framework is disposing the platform view. Implement this to properly dispose resources of the new platform view from the platform.
+  ///
+  /// e.g.
+  /// ```dart
+  /// Future<void> dispose() async {
+  ///   await SystemChannels.platform_views.invokeMethod<void>('dispose', id);
+  /// }
+  /// ```
+  /// See `SystemChannels.platform_views` for more details.
+  void dispose();
```

Comment:
I'm not clear on why this class is needed.  The only thing a PlatformViewController does is dispose things?  Implementers would not have to implement anything else?

Reply:
There will be more added in later PRs. e.g. clearFocus and dispatchPointerEvents. As they are not related to this PR so I didn't add them.

Reply:
It will also have clearFocus and dispatchPointerEvents in later PRs. Since these are not a part of this PR, so i didn't add them.
Thinking about it, dispose is also not used in this PR. Maybe I'll remove dispose in this PR as well?
Then it leaves an empty abstract class that only is required to notify the renderObject to markNeedsPaint. It seems odd and confusing as you pointed out.
cc @amirh 

The other solution would be to add the `PlatformViewLink` to this patch to perform the dispose so the value of `PlatformViewController` can be clearly seen. Then we are locked down to the name `PlatformViewLink`.

Reply:
I'm fine with leaving it with no methods as an incremental step to keep PRs at reasonable size given that the next PR is going to add stuff here.

Though maybe we should be adding an id getter here (see comment above).

Reply:
I would also need a setter right? I added a variable in `PlatformViewController` for now. Maybe I should make a setter to make it only allowed to be set once?